pkg/signup/service/signup_service.go

```diff
@@ -223,17 +233,22 @@ func (s *ServiceImpl) reactivateUserSignup(ctx *gin.Context, existing *toolchain
 // GetSignup returns Signup resource which represents the corresponding K8s UserSignup
 // and MasterUserRecord resources in the host cluster.
 // Returns nil, nil if the UserSignup resource is not found or if it's deactivated.
-func (s *ServiceImpl) GetSignup(userID string) (*signup.Signup, error) {
-
-	// Retrieve UserSignup resource from the host cluster
-	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Get(EncodeUserID(userID))
+func (s *ServiceImpl) GetSignup(userID, username string) (*signup.Signup, error) {
+	// Retrieve UserSignup resource from the host cluster, using the specified UserID and username
+	userSignup, err := s.GetUserSignup(userID, username)
+	// If an error was returned, then return here
 	if err != nil {
 		if apierrors.IsNotFound(err) {
```

Comment:
if the `UserSignup` is not found when looking-up by `userID`, then we shouldn't return, but instead we should look-up by `username`, shouldn't we?

Reply:
This is all handled in the `GetUserSignup()` function - if the lookup by `userID` fails, then it attempts to find the `UserSignup` again using the `username` instead.

Reply:
ah my bad, I missread the parameters in the `GetUserSignup` method 🤦‍♂️